scripts/monitoring/cron-send-ssl-check.py

```diff
@@ -0,0 +1,142 @@
+    logger.info("oc command took %s seconds", str(time.time() - oc_time))
+    return oc_result
+
+def runOCcmd_yaml(cmd, base_cmd='oc'):
+    """ log commands through ocutil """
+    logger.info(base_cmd + " " + cmd)
+    ocy_time = time.time()
+    ocy_result = ocutil.run_user_cmd_yaml(cmd, base_cmd=base_cmd, )
+    logger.info("oc command took %s seconds", str(time.time() - ocy_time))
+    return ocy_result
+
+def parse_args():
+    """ parse the args from the cli """
+    logger.debug("parse_args()")
+
+    parser = argparse.ArgumentParser(description='OpenShift public ssl expired info check ')
+    parser.add_argument('-v', '--verbose', action='store_true', default=None, help='Verbose?')
+    return parser.parse_args()
+
+def send_metrics(day_left):
+    """ send data to MetricSender"""
+    logger.debug("send_metrics()")
+
+    ms_time = time.time()
+    ms = MetricSender()
+    logger.info("Send data to MetricSender")
+    ms.add_metric({'openshift.master.public.ssl.left': day_left})
+    ms.send_metrics()
+    logger.info("Data sent to Zagg in %s seconds", str(time.time() - ms_time))
+
+
+
+def get_ssl_info(domain_name):
+    """get the domain expired date"""
+    cert = ssl.get_server_certificate((domain_name, 443))
+    x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
+
+    notafter = x509.get_notAfter()
+    notafter = notafter[0:8]
+    now = datetime.datetime.now()
+    notafter_t = datetime.datetime.strptime(notafter, '%Y%m%d')
+    delta = notafter_t - now
+    logger.info("public url "+domain_name+"will expired in :"+ str(delta.days))
+    return delta.days
+
+def main():
+    """ check ssl expired info """
+
+    logger.info('################################################################################')
+    logger.info('start check the ssl info on this cluster')
+    logger.info('################################################################################')
+
+    args = parse_args()
+
+    if args.verbose:
+        logger.setLevel(logging.DEBUG)
+
+    urls = ['console', 'api', 'registry', 'logs', 'metrics']
+
+    ############# setup() #############
+    try:
+        #get registry console for url with shard id check
+        registry_info = runOCcmd_yaml("get route/registry-console -n default")
+        registry_domain = registry_info['spec']['host']
+        expire_day_small = get_ssl_info(registry_domain)
+        #get clusterid
+        clusterid = registry_info['spec']['host'].split('.')[2]
+
+        for url in urls:
+            url_public = url + "." + clusterid + '.openshift.com'
+            expire_day_left = get_ssl_info(url_public)
+            if expire_day_small > expire_day_left:
+                expire_day_small = expire_day_left
+        #return the smallest day on this cluster
+    except Exception as e:
+        logger.exception("error during check the ssl expired info")
+        exception = e
+    try:
+        send_metrics(expire_day_small)
```

Comment:
it is possible that `send_metrics` is called without `expire_day_small` being set. Please fix this.